### PR TITLE
Implementing null-aware operators throughout the repository

### LINKED ISSUE #143803: Proposal to implement null-aware operators (`?.` and `??`) throughout the repo
Most of the Flutter codebase takes full advantage of null-aware operators, but I found a few spots here and there that could be improved.

```dart
// before
if (query != null) {
  return query._data;
}
return null;

// after
return query?.data;
```

```dart
// before
if (box != null) {
  return box.size.width;
}
return _kWidth;

// after
return box?.size.width ?? _kWidth;
```

```dart
// before
if (visual != null) {
  return visual;
}
if (logical != null) {
  return logical;
}
return 'BorderRadius.zero';

// after
return visual ?? logical ?? 'BorderRadius.zero';
```

### PR DESCRIPTION
**Pull request by nate-thegrate**
This pull request fixes #143803 by taking advantage of Dart's null-aware operators.

And unlike `switch` expressions ([9 PRs](https://github.com/flutter/flutter/pull/143634) and counting), the Flutter codebase is already fantastic when it comes to null-aware coding. After refactoring the entire repo, all the changes involving `?.` and `??` can fit into a single pull request.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/drawer.dart [hunk 1]
```diff
@@ -560,10 +560,8 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
 
   double get _width {
     final RenderBox? box = _drawerKey.currentContext?.findRenderObject() as RenderBox?;
-    if (box != null) {
-      return box.size.width;
-    }
-    return _kWidth; // drawer not being shown currently
+    // set to _kWidth if drawer not being shown currently
```
Comment (justinmc) [inline packages/flutter/lib/src/material/drawer.dart:563]: Nit: "Return _kWidth if drawer not being shown currently."

File: packages/flutter_tools/lib/src/commands/packages.dart [hunk 2]
```diff
@@ -399,14 +399,7 @@ class PackagesGetCommand extends FlutterCommand {
...
 
-  late final String? _androidEmbeddingVersion = (() {
-    final FlutterProject? rootProject = _rootProject;
-    if (rootProject == null) {
-      return null;
-    }
-
-    return rootProject.android.getEmbeddingVersion().toString().split('.').last;
-  })();
+  late final String? _androidEmbeddingVersion = _rootProject?.android.getEmbeddingVersion().toString().split('.').last;
```
Comment (justinmc) [inline packages/flutter_tools/lib/src/commands/packages.dart:402]: I really wonder why this was done in a self calling function in the first place... Maybe just to limit the scope of the variables?

**Review by justinmc — APPROVED** [review]
LGTM 👍. 200+ lines saved! Thanks for all of these code migrations. Out of curiosity are you using scripts or regexes or something to help do this?

**Comment by nate-thegrate** [comment]
> LGTM 👍. 200+ lines saved! Thanks for all of these code migrations. Out of curiosity are you using scripts or regexes or something to help do this?

Thanks very much! I've just been using VS Code's built-in regex search to look for potential stuff to improve.  
For this PR it was `== null` or `!= null` with something being assigned or returned in the next line.

**Comment by justinmc** [comment]
Ah got it. Maybe at some point we can add analyzer/lint rules for stuff like this.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by Juliotati** [comment]
**BEAUTIFUL**

**Pull request merged**